### PR TITLE
feat: add avatar and colored avatar

### PR DESCRIPTION
**Pull request by SimBig**

### CODE REVIEW

**Comment by spawnia** [comment]
CI is failing.

File: src/Avatar/utils.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  return `hsla(${h}, ${s}%, ${l}%,${opacity})`;
+}
+
+function range(hash: number, min: any, max: any) {
+  const diff = max - min;
+  const x = ((hash % diff) + diff) % diff;
+  return x + min;
+}
+
+export function stringToHexColor(string: string): string {
```
Comment (spawnia) [inline src/Avatar/utils.ts:26]: Needs testing, preferably some kind of property based test. Things that need to be true about this function:
- Various inputs always output a correctly formatted hex code
- The hex code of various inputs differs
- It is pure

**Review by spawnia — CHANGES_REQUESTED** [review]

File: src/Avatar/utils.ts (hunk 1 above)
Comment (SimBig) [reply src/Avatar/utils.ts]: added tests

File: src/Avatar/utilis.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    let lastResult = '';
+    testStrings.forEach((testString: string) => {
+      const newResult = stringToHslaColor(testString);
+      expect(newResult).not.toEqual(lastResult);
+      lastResult = newResult;
+    });
+  });
+
+  it('returns a valid color for a random string', () => {
+    expect(checkIsValidColor(stringToHslaColor(randomString))).toBeTruthy();
```
Comment (spawnia) [inline src/Avatar/utilis.test.tsx:26]:
```suggestion
    expect(checkIsValidColor(stringToHslaColor(randomString))).toBe(true);
```

File: src/Avatar/utilis.test.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  });
+});
+
+function checkIsValidColor(color: string): boolean {
```
Comment (spawnia) [inline src/Avatar/utilis.test.tsx:30]:
```suggestion
function isValidColor(color: string): boolean {
```

File: src/Avatar/utils.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+function getHslColor(hash: number) {
+  const h = range(hash, 0, 360);
+  const s = range(hash, 75, 80);
+  const l = range(hash, 40, 70);
+  const opacity = '0.9';
+
+  return `hsla(${h}, ${s}%, ${l}%, ${opacity})`;
+}
+
+function range(hash: number, min: any, max: any) {
```
Comment (spawnia) [inline src/Avatar/utils.ts:20]:
```suggestion
function range(hash: number, min: number, max: number): number {
```

File: src/Avatar/utils.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+function hashCode(string: string) {
```
Comment (spawnia) [inline src/Avatar/utils.ts:1]:
```suggestion
function hashCode(string: string): number {
```

File: src/Avatar/utils.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  let hash = 0;
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < string.length; i++) {
+    // eslint-disable-next-line no-bitwise
+    hash = string.charCodeAt(i) + ((hash << 5) - hash);
+  }
+  return hash;
+}
+
+function getHslColor(hash: number) {
```
Comment (spawnia) [inline src/Avatar/utils.ts:11]:
```suggestion
function getHslColor(hash: number): string {
```

File: src/Avatar/utilis.test.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    'is a pure function returning the same hsla for the same string',
+    (testString) => {
+      expect(stringToHslaColor(testString)).toStrictEqual(
+        'hsla(58, 78%, 68%, 0.9)',
+      );
+    },
+  );
+
+  it('returns various hsla colors for various inputs', () => {
+    const testStrings = ['t', 'te', 'tes', 'test', randomString];
```
Comment (spawnia) [inline src/Avatar/utilis.test.tsx:16]: Does not hold true if `randomString` is exactly `'test'`.

File: src/Avatar/utilis.test.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    let lastResult = '';
```
Comment (spawnia) [inline src/Avatar/utilis.test.tsx:17]:
```ts
import { uniq } from 'lodash';

expect(uniq(testStrings.map(stringToHslaColor)).length).toBe(testStrings.length);
```

**Review by spawnia — CHANGES_REQUESTED** [review]

File: src/Avatar/utilis.test.tsx (hunk 7 above)
Comment (SimBig) [reply src/Avatar/utilis.test.tsx:16]: I think the test is still valid and good. what would you do about it? Leave it out at all?

**Review by spawnia — APPROVED** [review]

**Pull request merged**